lib/offsite_payments/integrations/poli_pay.rb

```diff
@@ -0,0 +1,247 @@
+module OffsitePayments
+  module Integrations
+    module PoliPay
+      def self.notification(post, options = {})
+        Notification.new(post, options)
+      end
+
+      def self.base_url
+        "https://poliapi.apac.paywithpoli.com/api"
+      end
+
+      def self.token_url
+        "#{base_url}/Transaction/Initiate"
+      end
+
+      def self.query_url(token)
+        "#{base_url}/Transaction/GetTransaction?token=#{CGI.escape(token)}"
+      end
+
+      def self.return(query_string, options = {})
+        Return.new(query_string, options)
+      end
+
+      def self.sign(fields, key)
+        OpenSSL::HMAC.hexdigest(OpenSSL::Digest::SHA256.new, key, fields.sort.join)
+      end
+
+      class Interface
+        include ActiveUtils::PostsData # ssl_get/post
+
+        def initialize(login, password)
+          @login = login
+          @password = password
+        end
+
+        def query(token)
+          raise ArgumentError, "Token must be specified" if token.blank?
+          url = self.class.parent.query_url(token)
+          raw_response = ssl_get(url, standard_headers)
+          parse_response(raw_response)
+        rescue ActiveUtils::ResponseError => e
+          RequestError.new(:token, e)
```

Comment:
We return an error object in this case? That's quite inconsistent with the expected return type being a hash from parse_response